Simplified/Network/NYPLNetworkExecutor.swift

```diff
@@ -67,21 +67,127 @@ class NYPLNetworkExecutor {
-           completion: @escaping (_ result: NYPLResult<Data>) -> Void) {
+           completion: @escaping (_ result: NYPLResult<Data>) -> Void) -> URLSessionDataTask {
     let task = urlSession.dataTask(with: req)
     responder.addCompletion(completion, taskID: task.taskIdentifier)
     task.resume()
+    return task
   }
 }
 
 extension NYPLNetworkExecutor {
-  private func request(for url: URL) -> URLRequest {
-    return URLRequest(url: url,
-                      cachePolicy: urlSession.configuration.requestCachePolicy)
-  }
+    func request(for url: URL) -> URLRequest {
+
+        var urlRequest = URLRequest(url: url,
+                             cachePolicy: urlSession.configuration.requestCachePolicy)
+
+        if let authToken = NYPLUserAccount.sharedAccount().authToken {
+            let headers = [
+                "Authorization" : "Bearer \(authToken)",
+                "Content-Type" : "application/json"
+            ]
+
+            headers.forEach { urlRequest.setValue($0.value, forHTTPHeaderField: $0.key) }
+        }
+
+        return urlRequest
+    }
 
   func clearCache() {
     urlSession.configuration.urlCache?.removeAllCachedResponses()
   }
 }
+
+// Objective-C compatibility
+extension NYPLNetworkExecutor {
+    @objc class func bearerAuthorized(request: URLRequest) -> URLRequest {
+        var request = request
+        var headers: [String: String] = ["Authorization" : "",
+                                         "Content-Type" : "application/json"]
+        if let authToken = NYPLUserAccount.sharedAccount().authToken {
+            let authenticationValue = "Bearer \(authToken)"
+            headers = ["Authorization" : "\(authenticationValue)",
+                "Content-Type" : "application/json"]
+        }
+        for (headerKey, headerValue) in headers {
+            request.setValue(headerValue, forHTTPHeaderField: headerKey)
+        }
+        return request
+    }
+
+    /// Performs a GET request using the specified URL
+    /// - Parameters:
+    ///   - reqURL: URL of the resource to GET.
+    ///   - completion: Always called when the resource is either fetched from
+    /// the network or from the cache.
+    @objc func download(_ reqURL: URL,
+                   completion: @escaping (_ result: Data?, _ response: URLResponse?,  _ error: Error?) -> Void) -> URLSessionDownloadTask {
+        let req = request(for: reqURL)
+        let completionWrapper: (_ result: NYPLResult<Data>) -> Void = { result in
+            switch result {
+            case let .success(data, response): completion(data, response, nil)
+            case let .failure(error, response): completion(nil, response, error)
+            }
+        }
+
+        let task = urlSession.downloadTask(with: req)
+        responder.addCompletion(completionWrapper, taskID: task.taskIdentifier)
+        task.resume()
+        
+        return task
+    }
+
+    /// Performs a GET request using the specified URL
```

Comment:
is this for executing a generic URLRequest?

Reply:
yes, exactly, a wrapper for your executeRequest which can be used in objc (it wouldn't handle enum with associated values) :)

Reply:
got it -- I would maybe rename it to `execute` to keep the terminology consistent?